Services dhcp relay

Bouwdie opened the pull request.

Bouwdie committed:
Ran clean.sh.

Bouwdie committed:
Merge branch 'bootstrap' of https://github.com/Bouwdie/pfsense into services_dhcp_relay

Caught up with main repo.

Bouwdie committed:
Added fix in IpAddress to check if a mask is set to prevent a method call on a non object. 
Migrated form. 
Moved server input to an array form value.

SjonHortensius commented on usr/local/www/services_dhcp_relay.php:
```diff
@@ -106,106 +125,73 @@
-	}
+if ($dhcpd_enabled)
+{
+	echo "DHCP Server is currently enabled. Cannot enable the DHCP Relay service while the DHCP Server is enabled on any interface.";
```
Could you wrap this in an alert? http://getbootstrap.com/components/#alerts

Bouwdie replied:
I was thinking about it, wrapped in in the print_error_messages first. Maybe the messaging system can be abstracted.

SjonHortensius commented on usr/local/www/services_dhcp_relay.php:
```diff
@@ -106,106 +125,73 @@
+$form = new Form;
+
+$section = new Form_Section('DHCP Relay configuration');
+$form->add($section);
```
I'm a bit surprised this works, can you move the add to be after all add() calls have been done? Eg. just before `print $form`

Bouwdie replied:
Why wouldn't it work? You work on the same object reference. For clarity i shall move it down.

SjonHortensius commented on usr/local/www/services_dhcp_relay.php:
```diff
@@ -40,13 +40,29 @@
+function filterDestinationServers(array $destinationServers)
+{
+	return array_unique(
+		array_filter(
```
I actually think if you're using array_filter with (!empty), you can remove the callback entirely, as the default of array_filter is remove everything that evals to false (which all empties do)

Bouwdie committed:
Use default array_filter callback. 
Moved adding of section down for clarity. 
Wrapped message in an alert (same as print_input errors but without the extra txt).

Bouwdie commented:
I see that services_dhcpv6_relay.php is almost the same as this form. Do we combine these or keep duplicated forms?

SjonHortensius commented:
You can definitely combine them, just check if/where the v6 variant is
referenced in existing files
